File: C/daily-practice/day7.1/src/main.c
#include <stdio.h> 

int main() {
    char hello[] = "hello", s[101] = {};
    int hIndex = 0;
    scanf("%100s", s);
    printf("%s\n%s", s, hello);
    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == hello[hIndex]) {
            hIndex++;
        }
        if (hIndex > 4) {
            printf("YES\n");
            return 0;
        }
    }
    printf("NO\n");
    return 0;
}

File: go/daily-practice/day7.1/main.go
package main

import "fmt"

var (
    hello = []rune{'h', 'e', 'l', 'l', 'o'}
    word string
    pos int
)

func main() {
    fmt.Scan(&word)
    for _, char := range []rune(word) {
        if char == hello[pos] {
            pos++
            if pos > 4 {
                fmt.Println("YES")
                return
            }
        }
    }
    fmt.Println("NO")
}
